tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/arena/ArenaContextSync.scala

```diff
@@ -0,0 +1,373 @@
+  def getDom(funCell: ArenaCell): ArenaCell = arena.getDom(funCell)
+
+  /**
+   * Get the co-domain cell associated with a function.
+   *
+   * @param funCell
+   *   a function cell.
+   * @return
+   *   the co-domain cell
+   */
+  def getCdm(funCell: ArenaCell): ArenaCell = arena.getCdm(funCell)
+
+  /**
+   * Check, whether a cell has an associated domain edge.
+   *
+   * @param cell
+   *   a cell
+   * @return
+   *   true, if cell has an edge labelled with 'dom'
+   */
+  def hasDom(cell: ArenaCell): Boolean = arena.hasDom(cell)
+
+  /**
+   * Check, whether a cell has an associated co-domain edge.
+   *
+   * @param cell
+   *   a cell
+   * @return
+   *   true, if cell has an edge labelled with 'cdm'
+   */
+  def hasCdm(cell: ArenaCell): Boolean = arena.hasCdm(cell)
+
+  /**
+   * Check, whether two cells are connected with a 'has' edge.
+   *
+   * @param src
+   *   an edge origin
+   * @param dst
+   *   an edge destination
+   * @return
+   *   true, if src has an edge to dst labelled with 'has'
+   */
+  def isLinkedViaHas(src: ArenaCell, dst: ArenaCell): Boolean = arena.isLinkedViaHas(src, dst)
+
+  /**
+   * Find all the cells of a given type.
+   *
+   * @return
+   *   all the cells that have exactly the same type as the argument (no unification involved)
+   */
+  def findCellsByType(cellType: CellT): Seq[ArenaCell] = arena.findCellsByType(cellType)
+
+  /**
+   * Print the graph structure induced by 'has' edges starting with root.
+   *
+   * @param root
+   *   a cell to start from
+   */
+  def subgraphToString(root: ArenaCell): String = {
+    val builder = new StringBuilder()
+
+    def print(cell: ArenaCell): Unit = {
+      builder.append(cell.id)
+      builder.append(" ")
+      val cells = getHas(cell)
+      if (cells.nonEmpty) {
+        builder.append("has:(")
+        for (c <- cells)
+          print(c.elem)
+        builder.append(")")
+      }
+    }
+
+    print(root)
+    builder.mkString
+  }
+
+}
+
+object ArenaContextSync {
+  def create(solverContext: SolverContext): ArenaContextSync = {
+    val emptyArena = PureArena.empty
+    // by convention, the first cells have the following semantics:
+    //  0 stores FALSE, 1 stores TRUE, 2 stores BOOLEAN, 3 stores Nat, 4 stores Int
+    val appendSeq = Seq(
+        CellTFrom(BoolT1),
+        CellTFrom(BoolT1),
+        CellTFrom(SetT1(BoolT1)),
+        InfSetT(CellTFrom(IntT1)),
+        InfSetT(CellTFrom(IntT1)),
+    )
+
+    val (initArena, cells) = appendSeq.foldLeft((emptyArena, Seq.empty[ArenaCell])) { case ((arena, seq), cellT) =>
+      val newCell = arena.nextCell(cellT)
+      (arena.appendCell(newCell), seq :+ newCell)
+    }
+    // declare Boolean cells in SMT
+    val Seq(cellFalse, cellTrue, cellBoolean, cellNat, cellInt) = cells
+
+    solverContext.declareCell(cellFalse)
+    solverContext.declareCell(cellTrue)
+    solverContext.declareCell(cellBoolean)
+    solverContext.declareCell(cellNat)
+    solverContext.declareCell(cellInt)
```

Comment:
why not make a for-loop over `cells`?

Reply:
Yeah, makes sense, I guess I was on autopilot here.